From 9248331f1b0b1477d1dc248d934cf0e2adbe6cd2 Mon Sep 17 00:00:00 2001
From: Chenyang Zhong <zhongcy95@gmail.com>
Date: Thu, 3 Jun 2021 07:58:22 -0400
Subject: [PATCH] qcacld-3.0: fix the BSS check

The OR logic always evaluates to true, which obviously is not
intended to happen. Fix it by changing the logic to AND.

Signed-off-by: Chenyang Zhong <zhongcy95@gmail.com>
Change-Id: Ic63f68258793254493123875ffa90b2909eeb7e0
Signed-off-by: Tashfin Shakeer Rhythm <tashfinshakeerrhythm@gmail.com>
---
 drivers/staging/qcacld-3.0/core/sme/src/csr/csr_api_roam.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/qcacld-3.0/core/sme/src/csr/csr_api_roam.c b/drivers/staging/qcacld-3.0/core/sme/src/csr/csr_api_roam.c
index f7164f231239..0f476de670d2 100644
--- a/drivers/staging/qcacld-3.0/core/sme/src/csr/csr_api_roam.c
+++ b/drivers/staging/qcacld-3.0/core/sme/src/csr/csr_api_roam.c
@@ -4573,8 +4573,8 @@ QDF_STATUS csr_roam_prepare_bss_config(struct mac_context *mac,
 	if (((pBssConfig->uCfgDot11Mode == eCSR_CFG_DOT11_MODE_11N)
 	    || (pBssConfig->uCfgDot11Mode == eCSR_CFG_DOT11_MODE_11AC))
 		&& ((pBssConfig->qosType != eCSR_MEDIUM_ACCESS_WMM_eDCF_DSCP)
-		    || (pBssConfig->qosType != eCSR_MEDIUM_ACCESS_11e_HCF)
-		    || (pBssConfig->qosType != eCSR_MEDIUM_ACCESS_11e_eDCF))) {
+		    && (pBssConfig->qosType != eCSR_MEDIUM_ACCESS_11e_HCF)
+		    && (pBssConfig->qosType != eCSR_MEDIUM_ACCESS_11e_eDCF))) {
 		/* Joining BSS is 11n capable and WMM is disabled on AP. */
 		/* Assume all HT AP's are QOS AP's and enable WMM */
 		pBssConfig->qosType = eCSR_MEDIUM_ACCESS_WMM_eDCF_DSCP;
-- 
2.49.0

